controller_manager:
  ros__parameters:
    update_rate: 100  # Increased from default
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

joint_state_broadcaster:
  ros__parameters: {}

joint_trajectory_controller:
  ros__parameters:
    joints: [joint1, joint2, joint3, joint4, joint5, joint6, joint7, joint8,
             joint9, joint10, joint11, joint12, joint13, joint14, joint15,
             joint16, joint17, joint18, joint19, joint20, joint21, joint22,
             joint23, joint24, joint25, joint26, joint27, joint28, joint29,
             joint30, joint31, joint32, joint33]
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    constraints:
      goal_time: 0.5
      stopped_velocity_tolerance: 0.01
    gains:  # PID gains for position control
      joint1: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint2: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint3: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint4: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint5: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint6: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint7: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint8: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint9: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint10: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint11: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint12: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint13: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint14: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint15: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint16: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint17: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint18: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint19: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint20: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint21: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint22: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint23: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint24: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint25: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint26: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint27: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint28: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint29: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint30: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint31: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint32: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
      joint33: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 1.0}
    hold_trajectory_duration: 0.1
    velocity_ff_gain: 1.0
    state_publish_rate: 100.0
    action_monitor_rate: 20.0
